packages/tools/kadena-cli/src/account/utils/validateAccountDetails.ts

```diff
@@ -0,0 +1,61 @@
+import type { CommandResult } from '../../utils/command.util.js';
+import type {
+  IAccountDetailsResult,
+  IAddAccountManualConfig,
+} from '../types.js';
+import { compareConfigAndAccountDetails } from './compareConfigAndAccountDetails.js';
+import { createAccountName } from './createAccountName.js';
+import { getAccountDetailsFromChain } from './getAccountDetails.js';
+
+export interface IValidateAccountDetailsData {
+  isConfigAreSame: boolean;
+  config: IAddAccountManualConfig;
+  accountDetails: IAccountDetailsResult;
+}
+
+export async function validateAccountDetails(
+  config: IAddAccountManualConfig,
+): Promise<CommandResult<IValidateAccountDetailsData>> {
+  try {
+    const accountName =
+      config.accountName === undefined
+        ? await createAccountName(config)
+        : config.accountName;
+
+    const configWithAccountName = {
+      ...config,
+      accountName,
+    };
+
+    const accountDetails = await getAccountDetailsFromChain(
+      configWithAccountName,
+    );
+
+    const isConfigAreSame = compareConfigAndAccountDetails(
+      configWithAccountName,
+      accountDetails,
+    );
+
+    return {
+      success: true,
+      data: {
+        config: configWithAccountName,
+        accountDetails,
+        isConfigAreSame,
+      },
+    };
+  } catch (error) {
+    if (error.message.includes('row not found') === true) {
```

Comment:
if you add an account that does not exist on chain, you should still write the config to a file and remember the account, but maybe inform the user that it does not exist on-chain.